pkg/controller/workloaddispatcher/incrementaldispatcher_test.go

```diff
@@ -0,0 +1,289 @@
+				ac := utiltesting.MakeAdmissionCheck(string(tc.mkAcState.Name)).
+					ControllerName(kueue.MultiKueueControllerName).
+					Parameters(kueue.GroupVersion.Group, "MultiKueueConfig", string(tc.mkAcState.Name)).
+					Obj()
+
+				objs = append(objs, ac)
+			}
+
+			if tc.workload != nil {
+				objs = append(objs, tc.workload)
+			}
+
+			if tc.mkAcState != nil {
+				mkConfig := utiltesting.MakeMultiKueueConfig(string(tc.mkAcState.Name)).Clusters("cluster1").Obj()
+				objs = append(objs, mkConfig)
+			}
+			scheme := runtime.NewScheme()
+			utilruntime.Must(kueue.AddToScheme(scheme))
+
+			if tc.clusters != nil {
+				for _, cluster := range tc.clusters {
+					objs = append(objs, &cluster)
+				}
+			}
+			cl := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(objs...).Build()
+			helper, _ := admissioncheck.NewMultiKueueStoreHelper(cl)
+			rec := &IncrementalDispatcherReconciler{
+				client:          cl,
+				helper:          helper,
+				clock:           fakeClock,
+				dispatcherName:  tc.dispatcherName,
+				roundStartTimes: make(map[types.NamespacedName]time.Time),
+			}
+
+			req := ctrl.Request{NamespacedName: types.NamespacedName{Name: workloadName, Namespace: testNamespace}}
+			ctx, _ := utiltesting.ContextWithLog(t)
+			_, gotErr := rec.Reconcile(ctx, req)
+			if diff := cmp.Diff(tc.wantErr, gotErr); diff != "" {
+				t.Errorf("Unexpected error (-want/+got)\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestIncrementalDispatcherNominateWorkers(t *testing.T) {
+	const (
+		testNamespace = "default"
+		testName      = "test-wl"
+	)
+
+	testCases := []struct {
+		name               string
+		remoteCount        int
+		preNominated       []string
+		expectNominated    int
+		wantErr            error
+		advanceRoundTime   bool
+		expectNominatedSet []string
+	}{
+		{"one remote", 1, nil, 1, nil, false, []string{"A"}},
+		{"two remotes", 2, nil, 2, nil, false, []string{"A", "B"}},
+		{"three remotes", 3, nil, 3, nil, false, []string{"A", "B", "C"}},
+		{"fifteen remotes", 15, nil, 3, nil, false, []string{"A", "B", "C"}},
+		{"all already nominated (3)", 3, []string{"A", "B", "C"}, 3, ErrNoMoreWorkers, true, []string{"A", "B", "C"}},
+		{"all already nominated (1)", 1, []string{"A"}, 1, ErrNoMoreWorkers, true, []string{"A"}},
+		{"round expired, next set nominated", 6, []string{"A", "B", "C"}, 6, nil, true, []string{"A", "B", "C", "D", "E", "F"}},
+		{"round in progress, keep current", 6, []string{"A", "B", "C"}, 3, nil, false, []string{"A", "B", "C"}},
+		{"round expired, nominate all", 8, []string{"A", "B", "C", "D", "E", "F"}, 8, nil, true, []string{"A", "B", "C", "D", "E", "F", "G", "H"}},
+		{"no remotes", 0, nil, 0, ErrNoMoreWorkers, false, []string{}},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			remotes := sets.New[string]()
+			for i := 1; i <= tc.remoteCount; i++ {
+				name := string(rune('A' + i - 1))
+				remotes.Insert(name)
+			}
```

Comment:
Can you avoid for autogenerated test data? I would like to keep remote clusters.
So, my proposal is having `remoteClusters sets.Set[string]` field in `testCases`.